src/Aspire.Hosting/Dcp/DcpDependencyCheck.cs

```diff
@@ -0,0 +1,239 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Text.Json.Serialization;
+using Aspire.Hosting.Dcp.Process;
+using System.Text;
+using System.Text.Json;
+using System.Text.RegularExpressions;
+using System.Globalization;
+using Aspire.Hosting.Properties;
+
+namespace Aspire.Hosting.Dcp;
+
+internal sealed partial class DcpDependencyCheck
+{
+    // Docker goes to into resource saver mode after 5 minutes of not running a container (by default).
+    // While in this mode, the commands we use for the docker runtime checks can take quite some time
+    private const int WaitTimeForDcpInfoCommandInSeconds = 10;
+
+    [GeneratedRegex("[^\\d\\.].*$")]
+    private static partial Regex VersionRegex();
+
+    public static async Task EnsureDcpDependenciesAsync(string? dcpPath, bool hasContainers, string? containerRuntime, CancellationToken cancellationToken)
+    {
+        if (!File.Exists(dcpPath))
+        {
+            throw new FileNotFoundException($"The Aspire application host is not installed at \"{dcpPath}\". The application cannot be run without it.", dcpPath);
+        }
+
+        try
+        {
+            await EnsureDcpVersionAsync(dcpPath, cancellationToken).ConfigureAwait(false);
```

Comment:
I think the result of the "info" command contains the "version" info as well? If so, we should use a single invocation for DCP, this will minimize the app startup perf impact.

And yes, I know, the older versions did not have "info", but if that command fails, we sort of know we do not have the right version.

Reply:
I should read all the comments before I start replying to one of the threads 😀. I can switch back to that approach and treat it as version not supported.